src/scales.js

```diff
@@ -205,3 +206,24 @@ function coerceDate(x) {
     : x == null || isNaN(x = +x) ? undefined
     : new Date(x);
 }
+
+// prepare scales for exposure through the plot's scales() function
+export function exposeScales(scaleDescriptors) {
+  return function(key) {
+    if (registry.has(key))
+      return key in scaleDescriptors ? exposeScale(scaleDescriptors[key]) : undefined;
+    throw new Error(`no such scale ${key}`);
+  };
+}
+
+function exposeScale({scale, ...options}) {
+  for (const remove of ["domain", "range", "interpolate", "clamp", "round", "nice", "padding", "inset", "reverse", "family"]) delete options[remove];
```

Comment:
This will expose any field _not_ in this list but I think we want to be explicit about which fields are exposed, i.e., I’d prefer to invert the logic here and have an allowlist (default exclude) instead of a blocklist (default include).